dvc/tree/webhdfs.py

```diff
@@ -0,0 +1,143 @@
+import logging
+import os
+import threading
+from contextlib import contextmanager
+from urllib.parse import urlparse
+
+from funcy import cached_property, wrap_prop
+
+from dvc.hash_info import HashInfo
+from dvc.path_info import CloudURLInfo
+from dvc.progress import Tqdm
+from dvc.scheme import Schemes
+
+from .base import BaseTree
+
+logger = logging.getLogger(__name__)
+
+
+def update_pbar(pbar):
+    """Update pbar to accept the two arguments passed by hdfs"""
+
+    def update(_, bytes_transfered):  # No use for first argument (path)
+        if bytes_transfered == -1:
+            return
+        pbar.update(bytes_transfered)
+
+    return update
+
+
+class WebHDFSTree(BaseTree):
+    scheme = Schemes.WEBHDFS
+    PATH_CLS = CloudURLInfo
+    REQUIRES = {"hdfs": "hdfs"}
+    PARAM_CHECKSUM = "checksum"
+
+    def __init__(self, repo, config):
+        super().__init__(repo, config)
+
+        self.path_info = None
+        url = config.get("url")
+        if not url:
+            return
+
+        parsed = urlparse(url)
+        user = parsed.username or config.get("user")
+
+        self.path_info = self.PATH_CLS.from_parts(
+            scheme="webhdfs",
+            host=parsed.hostname,
+            user=user,
+            port=parsed.port,
+            path=parsed.path,
+        )
+
+        self.hdfscli_config = config.get("hdfscli_config")
+        self.token = config.get("webhdfs_token")
+        self.alias = config.get("webhdfs_alias")
+
+    @wrap_prop(threading.Lock())
+    @cached_property
+    def hdfs_client(self):
+        import hdfs
+
+        logger.debug("URL: %s", self.path_info)
+        logger.debug("HDFSConfig: %s", self.hdfscli_config)
+
+        try:
+            client = hdfs.config.Config(self.hdfscli_config).get_client(
+                self.alias
+            )
+        except hdfs.util.HdfsError:
+            # No hdfs config file was found, or parsing failed. Fallback to
+            # clients using Hadoop delegation token or username
+            http_url = f"http://{self.path_info.host}:{self.path_info.port}"
+
+            if self.token is not None:
+                client = hdfs.TokenClient(http_url, token=self.token, root="/")
+            else:
+                client = hdfs.InsecureClient(
+                    http_url, user=self.path_info.user, root="/"
+                )
```

Comment:
From https://github.com/iterative/dvc/pull/4747#discussion_r517441658 

You are still catching a generic error, is there a specific exc that is raised for the config error?

Reply:
I see what you mean. Not really, `hdfs` only wraps errors in `HDFSError`s, see: https://github.com/mtth/hdfs/blob/master/hdfs/config.py#L76

Reply:
@tomasfarias :( Ok, can we check the exc message just to make sure we are not misinterpreting the error?

Reply:
Gave it a shot. Really would have preferred different exception types since there's 3 potential error messages 😢